examples/remote-settings-cli/src/downloader.rs

```diff
@@ -0,0 +1,289 @@
+pub(crate) struct PackagedData {
+    pub(crate) collections: HashMap<String, CollectionData>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub(crate) struct CollectionData {
+    pub(crate) timestamp: u64,
+    pub(crate) records: Vec<Value>,
+}
+
+pub(crate) struct CollectionDownloader {
+    pub(crate) client: reqwest::Client,
+    pub(crate) multi_progress: Arc<MultiProgress>,
+    pub(crate) output_dir: PathBuf,
+}
+
+impl CollectionDownloader {
+    pub(crate) fn new(root_path: PathBuf) -> Self {
+        let output_dir = root_path.join(DEFAULT_OUTPUT_DIR);
+        Self {
+            client: reqwest::Client::new(),
+            multi_progress: Arc::new(MultiProgress::new()),
+            output_dir,
+        }
+    }
+
+    async fn fetch_collection(
+        &self,
+        collection: String,
+        pb: ProgressBar,
+    ) -> Result<(String, CollectionData)> {
+        // First get the last_modified from monitor
+        let monitor_url = format!(
+            "{}/buckets/monitor/collections/changes/records",
+            REMOTE_SETTINGS_URL,
+        );
+
+        let monitor_response: Value = self.client.get(&monitor_url).send().await?.json().await?;
+
+        // Find the last_modified for our collection
+        let last_modified = monitor_response["data"]
+            .as_array()
+            .context("No data array in monitor response")?
+            .iter()
+            .find(|record| {
+                record["collection"].as_str() == Some(&collection)
+                    && record["bucket"].as_str() == Some("main")
+            })
+            .context("Collection not found in monitor")?["last_modified"]
+            .as_u64()
+            .context("No last_modified in record")?;
+
+        // Now fetch the actual changeset
+        let url = format!(
+            "{}/buckets/main/collections/{}/changeset?_expected={}",
+            REMOTE_SETTINGS_URL, collection, last_modified
+        );
+
+        pb.set_message(format!("Downloading {}", collection));
+
+        let response = self
+            .client
+            .get(&url)
+            .send()
+            .await
+            .context(format!("Failed to fetch {}", collection))?;
+
+        let mut changeset: Value = response
+            .json()
+            .await
+            .context(format!("Failed to parse JSON for {}", collection))?;
+
+        // Special handling for search-config-v2
+        if collection == "search-config-v2" {
+            if let Some(changes) = changeset["changes"].as_array_mut() {
+                changes.sort_by(|a, b| {
+                    let a_type = a["recordType"].as_str().unwrap_or("");
+                    let b_type = b["recordType"].as_str().unwrap_or("");
+                    let a_id = a["identifier"].as_str().unwrap_or("");
+                    let b_id = b["identifier"].as_str().unwrap_or("");
+                    (a_type, a_id).cmp(&(b_type, b_id))
+                });
+            }
+        }
```

Comment:
Is this also sorting the properties within the records by name? (i.e. going through all the sub-objects within the records & sorting those properties as well). I can't see that it isn't, but would like to make sure. I believe `jq` does that for us in the mozilla-central output.

As a note, I'm thinking about expanding sorting to more of the search team's collections - the sorting helps with creating clean diffs in the repository. It'll probably be my next patch for `periodic_file_updates.sh`.

Reply:
I would like to pick this one up once we deal with `search-config-v2`. Based on the workflow, we might add collections as needed.